### ISSUE TITLE
Switching Network Type

### ISSUE THREAD

**Issue by PersonalityPi** [ElDewrito/DewRecode#16]
Switching to Online. Sometimes crashes the game. (Release build)

**Comment by PersonalityPi** [comment]
Also seems to happen even if you manage to host game. Not sure if this just a general HO problem or not. 80% sure it is but just mentioning it anyway.

**Comment by emoose** [comment]
Does this ever happen with ED at all?

I think it's related to the VoIP code in ChatPlugin, I noticed VoIP caused some crashes before but they've seemed to go away for me now, although sometimes when I switch to online and the VoIP text is being printed to the console http://i.imgur.com/X1HF8YF.png happens (where the text on the lobby menu just disappears)

If I disable the ChatPlugin it seems to work fine though. Maybe we'll have to recode the VoIP stuff, I noticed a lot of it seemed based on the sample TS3 code, it'd probably be better to recode it as part of the ModuleVoIP class instead.
Or it might be because I'm missing the soundbackends folder... I'll have to try copying it from my ED install now and see if it makes any difference.

Can you try renaming the ChatPlugin.dll to .dll_ and see if the crash still happens? Also do you have a soundbackends folder in your DR HO install?

**Comment by PersonalityPi** [comment]
Doesn't seem to happen in ED. The second one I  mentioned still happens but I think thats just related to HO itself rather then eldorito (It being hacky and all). 

I think the disappearing of menu text happened way before VOIP though. I think thats always been a general issue. I'm pretty sure that only started happening when DXHook was added in. 

Yeah I have the soundbackends folder in. I'll try renaming the plugin.

**Comment by PersonalityPi** [comment]
Yeah Didn't crash at all. With the DLL renamed. The crash is completely random though so I could have just been lucky.

However, I noticed if you tried to change the network type while the TS stuff was still loading up it would crash the game.
